### PR TITLE
Add GH action to close stale PRs

### LINKED ISSUE #7151: Enable GitHub Action to Close Inactive PRs
### Area for Improvement

Maintaining a clean and manageable PR queue is essential for project's maintainability. This proposal is to add a workflow to automatically close inactive PRs after a specified period. This will help declutter the PR queue and ensure that active contributions receive attention. A PR could land in this state due to many reasons, such as the submitter deciding not to continue working on the PR, changes in project direction making the PR irrelevant, or if the changes were included in another PR.

### Observed behavior

_No response_

### Desired behavior

Leverage the https://github.com/actions/stale GitHub action to automatically close PRs after 90 days of inactivity, with a notification period of 7 days. This will provide contributors with a reasonable window to address any outstanding feedback or issues before the PR is automatically closed. We can adjust this timeframe in the future based on the feedback and learnings. We can also consider PRs labeled with certain labels to be excluded from this workflow if we see the need for this in the future: https://github.com/actions/stale?tab=readme-ov-file#exempt-pr-labels.

### Proposed Fix

_No response_

### rad Version

Not applicable

### Operating system

_No response_

### Additional context

_No response_

### Would you like to support us?

- [ ] Yes, I would like to support you

[AB#11137](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/11137)

### PR DESCRIPTION
**Pull request by kachawla**
# Description

To declutter the PR queue and ensure that active contributions receive attention, enabling https://github.com/actions/stale github action to automatically close stale PRs after 90 days of inactivity, with a notification period of 7 days.

This will provide contributors with a reasonable window to address any outstanding feedback or issues before the PR is automatically closed. We can adjust this timeframe in the future based on the feedback and learnings. We can also consider PRs labeled with certain labels to be excluded from this workflow if we see the need for this in the future: https://github.com/actions/stale?tab=readme-ov-file#exempt-pr-labels.

## Type of change

- This pull request adds or changes features of Radius and has an approved issue (issue link required).
- This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Radius (issue link optional).

Fixes: https://github.com/radius-project/radius/issues/7151

### CODE REVIEW

File: .github/workflows/stale-prs.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    steps:
+    - uses: actions/stale@v5
+      with:
+        repo-token: ${{ secrets.GITHUB_TOKEN }}
+        stale-pr-message: 'This pull request has been automatically marked as stale because it has been inactive for 90 days. Remove stale label or comment or this PR will be closed in 7 days.'
+        stale-pr-label: 'stale'
+        days-before-pr-stale: 90 # 3 months
+        days-before-pr-close: 7
+        days-before-issue-stale: -1
+        days-before-issue-close: -1
```
Comment (kachawla) [inline .github/workflows/stale-prs.yml:27]: Disabling this action for issues.

**Review by ytimocin — DISMISSED** [review]

File: .github/workflows/stale-prs.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# This workflow warns and then closes PRs that have had no activity for 90 days.
+#
+# For more information, see:
+# https://github.com/actions/stale
+name: Close stale pull requests
```
[the remark this answers is not in the document]
Comment (kachawla) [reply .github/workflows/stale-prs.yml:5]: Yeah sure. I wanted to first make sure that as a team we are comfortable with this idea and then add docs. Will add and update the PR. Let me know if you have any questions/concerns about the workflow and values we are starting with.

[the remark this answers is not in the document]
Comment (kachawla) [reply .github/workflows/stale-prs.yml:5]: Added

File: .github/workflows/stale-prs.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+
+    steps:
+    - uses: actions/stale@v5
+      with:
+        repo-token: ${{ secrets.GITHUB_TOKEN }}
+        stale-pr-message: 'This pull request has been automatically marked as stale because it has been inactive for 90 days. Remove stale label or comment or this PR will be closed in 7 days.'
```
Comment (AaronCrawfis) [inline .github/workflows/stale-prs.yml:22]: Is there a way to opt-out of this behavior? What if we have a PR we know is on hold and we don't want it automatically closed? Maybe we can add a label to indicate that it shouldn't go stale?

Comment (AaronCrawfis) [reply .github/workflows/stale-prs.yml:22]: (although 3 months is a long time so removing the label won't be a daily thing. I think as designed this looks good)

Comment (kachawla) [reply .github/workflows/stale-prs.yml:22]: Yeah there is an option to opt out for specific labels https://github.com/actions/stale?tab=readme-ov-file#exempt-pr-labels, but I thought we should introduce it if really see the need for it.

**Review by AaronCrawfis — APPROVED** [review]
LGTM

File: .github/workflows/stale-prs.yml
Comment (ytimocin) [inline .github/workflows/stale-prs.yml:1]: nit: file can be named as **close-stale-prs.yml** maybe?

**Review by ytimocin — APPROVED** [review]

File: docs/contributing/contributing-pull-requests/README.md [hunk 4]
```diff
@@ -102,4 +102,8 @@ If you are the code reviewer, it's your responsibility to follow up (politely) i
 
 We welcome **any contributor or community member** to engage with **any pull request** on our repository. Feel free to make suggestions for improvements and ask questions that are relevant. If you're asking questions for your learning, please make it clear that your questions are "non-blocking" for the pull request.
 
-See the [code reviewing documentation](../contributing-code/contributing-code-reviewing/README.md) for guidance on code reviewing.
\ No newline at end of file
+See the [code reviewing documentation](../contributing-code/contributing-code-reviewing/README.md) for guidance on code reviewing.
+
+## Inactive Pull Requests
+
+Pull requests that have been inactive for 90 days will be marked with a stale label. They will automatically be closed after a subsequent 7 days of inactivity. This timeframe may be adjusted in the future based on project needs.
```
Comment (rynowak) [inline docs/contributing/contributing-pull-requests/README.md:109]: 👍

**Review by rynowak — APPROVED** [review]

File: .github/workflows/stale-prs.yml
Comment (kachawla) [reply .github/workflows/stale-prs.yml:1]: I agree, making it more explicit would be better. Sent a quick pr to update this: https://github.com/radius-project/radius/pull/7152

**Pull request merged**